llmfoundry/data/finetuning/tasks.py

```diff
@@ -102,6 +112,72 @@ def _is_empty_or_nonexistent(dirpath: str) -> bool:
     return not os.path.isdir(dirpath) or len(os.listdir(dirpath)) == 0
 
 
+def _get_role_key(message: Dict[str, str]) -> str:
+    if not isinstance(message, Mapping):
```

Comment:
this check now doesn't really make sense with the parameter typing. Should they just all be `Mapping`?

Reply:
"dict[str, str]" is incompatible with "Mapping[str, str]"

pyright does not know about dicts being mappings. We need a strict static check but a looser runtime check.